--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2246,7 +2246,9 @@ ADD_CUSTOM_TARGET(clang_tidy_prerequisites)
 ADD_DEPENDENCIES(clang_tidy_prerequisites protobuf_generated_all)
 ADD_DEPENDENCIES(clang_tidy_prerequisites GenError)

-ADD_SUBDIRECTORY(include)
+IF(NOT WITHOUT_CLIENTLIBS)
+  ADD_SUBDIRECTORY(include)
+ENDIF(NOT WITHOUT_CLIENTLIBS)
 ADD_SUBDIRECTORY(strings)
 ADD_SUBDIRECTORY(vio)
 ADD_SUBDIRECTORY(mysys)
@@ -2377,7 +2379,6 @@ ADD_SUBDIRECTORY(support-files)
 IF(NOT WITHOUT_SERVER)
   ADD_SUBDIRECTORY(mysql-test)
   ADD_SUBDIRECTORY(mysql-test/lib/My/SafeProcess)
-  ADD_SUBDIRECTORY(support-files)
   IF(WITH_INTERNAL)
     ADD_SUBDIRECTORY(internal)
   ENDIF()
--- a/libmysql/CMakeLists.txt
+++ b/libmysql/CMakeLists.txt
@@ -288,23 +288,32 @@   ADD_SUBDIRECTORY(authentication_win)
   LIST(APPEND LIBS_TO_MERGE auth_win_client)
 ENDIF()

-# LDAP authentication SASL client plug-in
-ADD_SUBDIRECTORY(authentication_ldap)
+IF(WITHOUT_CLIENTLIBS)
+  # Merge several convenience libraries into one big perconaserverclient
+  MERGE_CONVENIENCE_LIBRARIES(perconaserverclient ${LIBS_TO_MERGE}
+    COMPONENT Development
+    LINK_LIBRARIES ${LIBS_TO_LINK}
+    SKIP_INSTALL
+    )
+ELSE(WITHOUT_CLIENTLIBS)
+  # LDAP authentication SASL client plug-in
+  ADD_SUBDIRECTORY(authentication_ldap)

-# authentication kerberos client plug-in
-ADD_SUBDIRECTORY(authentication_kerberos)
+  # authentication kerberos client plug-in
+  ADD_SUBDIRECTORY(authentication_kerberos)

-# authentication IAM client plug-in
-ADD_SUBDIRECTORY(authentication_oci_client)
+  # authentication IAM client plug-in
+  ADD_SUBDIRECTORY(authentication_oci_client)

-# Fido and Webauthn clients
-ADD_SUBDIRECTORY(fido_client)
+  # Fido and Webauthn clients
+  ADD_SUBDIRECTORY(fido_client)

-# Merge several convenience libraries into one big perconaserverclient
-MERGE_CONVENIENCE_LIBRARIES(perconaserverclient ${LIBS_TO_MERGE}
-  COMPONENT Development
-  LINK_LIBRARIES ${LIBS_TO_LINK}
-  )
+  # Merge several convenience libraries into one big perconaserverclient
+  MERGE_CONVENIENCE_LIBRARIES(perconaserverclient ${LIBS_TO_MERGE}
+    COMPONENT Development
+    LINK_LIBRARIES ${LIBS_TO_LINK}
+    )
+ENDIF(WITHOUT_CLIENTLIBS)

 # Visual Studio users need debug  static library for debug projects
 IF(MSVC)
@@ -351,17 +360,32 @@ ENDIF()

 # Merge several convenience libraries into one big perconaserverclient
 # and link them together into shared library.
-MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
-  EXPORTS
-  ${CLIENT_API_FUNCTIONS}
-  ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
-  ${CLIENT_API_NONBLOCKING_FUNCTIONS}
-  COMPONENT SharedLibraries
-  LINK_LIBRARIES ${LIBS_TO_LINK}
-  ${UNIX_OUTPUT_NAME}
-  ${UNIX_SOVERSION}
-  ${UNIX_VERSION}
-  )
+IF(WITHOUT_CLIENTLIBS)
+  MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
+    EXPORTS
+    ${CLIENT_API_FUNCTIONS}
+    ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+    ${CLIENT_API_NONBLOCKING_FUNCTIONS}
+    COMPONENT SharedLibraries
+    LINK_LIBRARIES ${LIBS_TO_LINK}
+    ${UNIX_OUTPUT_NAME}
+    ${UNIX_SOVERSION}
+    ${UNIX_VERSION}
+    SKIP_INSTALL
+    )
+ELSE(WITHOUT_CLIENTLIBS)
+  MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
+    EXPORTS
+    ${CLIENT_API_FUNCTIONS}
+    ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+    ${CLIENT_API_NONBLOCKING_FUNCTIONS}
+    COMPONENT SharedLibraries
+    LINK_LIBRARIES ${LIBS_TO_LINK}
+    ${UNIX_OUTPUT_NAME}
+    ${UNIX_SOVERSION}
+    ${UNIX_VERSION}
+    )
+ENDIF(WITHOUT_CLIENTLIBS)

 # Downgrade warning for strncat in my_crypt_genhash.
 IF((WITH_LTO OR CMAKE_COMPILER_FLAG_WITH_LTO) AND MY_COMPILER_IS_GNU)
--- a/man/CMakeLists.txt
+++ b/man/CMakeLists.txt
@@ -26,7 +26,6 @@ SET(MAN1
   ibd2sdi.1
   innochecksum.1
   lz4_decompress.1
-  my_print_defaults.1
   myisam_ftdump.1
   myisamchk.1
   myisamlog.1
@@ -48,8 +47,6 @@ SET(MAN1
   mysqlpump.1
   mysqlshow.1
   mysqlslap.1
-  perror.1
-  zlib_decompress.1
   )

 IF(NOT WITH_SYSTEMD)
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -422,7 +422,7 @@ ELSE()
     ${CMAKE_CURRENT_BINARY_DIR}/${PKGCONFIG_FILE}
     ESCAPE_QUOTES @ONLY)

-  IF(INSTALL_PKGCONFIGDIR)
+  IF(INSTALL_PKGCONFIGDIR AND NOT WITHOUT_CLIENTLIBS)
     MESSAGE(STATUS "INSTALL ${PKGCONFIG_FILE} ${INSTALL_PKGCONFIGDIR}")
     INSTALL(FILES
       ${CMAKE_CURRENT_BINARY_DIR}/${PKGCONFIG_FILE}
@@ -435,23 +435,28 @@ ELSE()
   # mysqld_safe used in mtr even for systemd platforms
   IF(WITH_SYSTEMD)
     SET(BIN_SCRIPTS
-      mysql_config
       mysqldumpslow
       mysqld_safe
       ps_mysqld_helper
       ps-admin
       )
   ELSE()
     SET(BIN_SCRIPTS
-      mysql_config
       mysqldumpslow
       mysqld_multi
       mysqld_safe
       ps_mysqld_helper
       ps-admin
       )
   ENDIF()

+  IF(NOT WITHOUT_CLIENTLIBS)
+    SET(BIN_SCRIPTS
+      ${BIN_SCRIPTS}
+      mysql_config
+    )
+  ENDIF(NOT WITHOUT_CLIENTLIBS)
+
   FOREACH(file ${BIN_SCRIPTS})
     IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh)
       CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh
--- a/support-files/CMakeLists.txt
+++ b/support-files/CMakeLists.txt
@@ -24,8 +24,9 @@ IF(NOT UNIX)
   RETURN()
 ENDIF()

-INSTALL(FILES mysql.m4
-  DESTINATION ${INSTALL_SHAREDIR}/aclocal COMPONENT Development)
+IF(NOT WITHOUT_CLIENTLIBS)
+  INSTALL(FILES mysql.m4 DESTINATION ${INSTALL_SHAREDIR}/aclocal COMPONENT Development)
+ENDIF(NOT WITHOUT_CLIENTLIBS)

 SET(localstatedir "${MYSQL_DATADIR}")

--- a/utilities/CMakeLists.txt
+++ b/utilities/CMakeLists.txt
@@ -120,6 +120,31 @@ MYSQL_ADD_EXECUTABLE(static_thread_local_test
   EXCLUDE_FROM_ALL
   )

+IF(WITHOUT_CLIENTLIBS)
+MYSQL_ADD_EXECUTABLE(my_print_defaults
+  my_print_defaults.cc
+  COMPONENT Server
+  LINK_LIBRARIES mysys
+  SKIP_INSTALL )
+MYSQL_ADD_EXECUTABLE(perror
+  perror.cc
+  COMPONENT Server
+  DEPENDENCIES GenError
+  LINK_LIBRARIES mysys
+  SKIP_INSTALL )
+IF(WITH_LZ4 STREQUAL "bundled")
+  MYSQL_ADD_EXECUTABLE(lz4_decompress
+    lz4_decompress.cc
+    COMPONENT Server
+    LINK_LIBRARIES ext::lz4 mysys
+    SKIP_INSTALL )
+ENDIF()
+MYSQL_ADD_EXECUTABLE(zlib_decompress
+  zlib_decompress.cc
+  COMPONENT Server
+  LINK_LIBRARIES ext::zlib mysys
+  SKIP_INSTALL )
+ELSE(WITHOUT_CLIENTLIBS)
 MYSQL_ADD_EXECUTABLE(my_print_defaults
   my_print_defaults.cc
   COMPONENT Server
@@ -144,6 +169,7 @@
   COMPONENT Server
   LINK_LIBRARIES ext::zlib mysys
   )
+ENDIF(WITHOUT_CLIENTLIBS)

 # All targets below belong to COMPONENT Server and depend on InnoDB.
 IF(WITHOUT_SERVER)
