From 59b8e71442ba91c546171aac24c28b66dee08697 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Thu, 30 Apr 2020 20:01:20 +0200
Subject: [PATCH 4/7] tests: fix grant_user_lock test

Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
--- a/mysql-test/r/grant_user_lock.result
+++ b/mysql-test/r/grant_user_lock.result
@@ -98,13 +98,13 @@ CURRENT_USER()
 DROP USER ''@localhost;
 # Create anonymous user - LOCK
 CREATE USER ''@localhost IDENTIFIED BY 'pass' ACCOUNT LOCK;
-connect(localhost,,pass,test,MASTER_PORT,MASTER_SOCKET);
+connect(localhost, ,pass,test,MASTER_PORT,MASTER_SOCKET);
 ERROR HY000: Access denied for user '(null)'@'localhost'. Account is locked.
 DROP USER ''@localhost;
 # Disabling anonymous user
 CREATE USER ''@localhost IDENTIFIED BY 'pass';
 ALTER USER ''@localhost ACCOUNT LOCK;
-connect(localhost,,pass,test,MASTER_PORT,MASTER_SOCKET);
+connect(localhost, ,pass,test,MASTER_PORT,MASTER_SOCKET);
 ERROR HY000: Access denied for user '(null)'@'localhost'. Account is locked.
 DROP USER ''@localhost;
 # Enabling anonymous user
--- a/mysql-test/t/grant_user_lock.test
+++ b/mysql-test/t/grant_user_lock.test
@@ -112,7 +112,7 @@ DROP USER unlocked_user2@localhost;
 --echo # Create anonymous user
 connection default;
 CREATE USER ''@localhost IDENTIFIED BY 'pass';
-connect(anonymous_user_con, localhost, '', pass);
+connect(anonymous_user_con, localhost, ' ', pass);
 SELECT CURRENT_USER();
 disconnect anonymous_user_con;
 connection default;
@@ -121,7 +121,7 @@ DROP USER ''@localhost;
 --echo # Create anonymous user - explicit UNLOCK
 connection default;
 CREATE USER ''@localhost IDENTIFIED BY 'pass' ACCOUNT UNLOCK;
-connect(anonymous_user_con, localhost, '', pass);
+connect(anonymous_user_con, localhost, ' ', pass);
 SELECT CURRENT_USER();
 disconnect anonymous_user_con;
 connection default;
@@ -132,7 +132,7 @@ connection default;
 CREATE USER ''@localhost IDENTIFIED BY 'pass' ACCOUNT LOCK;
 --replace_result $MASTER_MYSOCK MASTER_SOCKET $MASTER_MYPORT MASTER_PORT
 --error ER_ACCOUNT_HAS_BEEN_LOCKED
-connect(anonymous_user_con, localhost, '', pass);
+connect(anonymous_user_con, localhost, ' ', pass);
 connection default;
 DROP USER ''@localhost;
 
@@ -142,7 +142,7 @@ CREATE USER ''@localhost IDENTIFIED BY 'pass';
 ALTER USER ''@localhost ACCOUNT LOCK;
 --replace_result $MASTER_MYSOCK MASTER_SOCKET $MASTER_MYPORT MASTER_PORT
 --error ER_ACCOUNT_HAS_BEEN_LOCKED
-connect(anonymous_user_con, localhost, '', pass);
+connect(anonymous_user_con, localhost, ' ', pass);
 connection default;
 DROP USER ''@localhost;
 
@@ -150,7 +150,7 @@ DROP USER ''@localhost;
 connection default;
 CREATE USER ''@localhost IDENTIFIED BY 'pass' ACCOUNT LOCK;
 ALTER USER ''@localhost ACCOUNT UNLOCK;
-connect(anonymous_user_con, localhost, '', pass);
+connect(anonymous_user_con, localhost, ' ', pass);
 SELECT CURRENT_USER();
 disconnect anonymous_user_con;
 connection default;
@@ -170,7 +170,7 @@ connection default;
 CREATE user ''@localhost IDENTIFIED BY 'pass';
 CREATE USER 'unlocked_user'@localhost IDENTIFIED BY 'pass';
 
-connect(anonymous_user_con, localhost, '', pass);
+connect(anonymous_user_con, localhost, ' ', pass);
 --error ER_TABLEACCESS_DENIED_ERROR
 UPDATE mysql.user SET account_locked='Y'
   WHERE user='unlocked_user' and host = 'localhost';
